Sources/Hilbert/py_hilbert.cc

```diff
@@ -215,16 +218,31 @@ void AddHilbertModule(py::module m) {
 
            ```
        )EOF")
-          .def("update_conf", &AbstractHilbert::UpdateConf, py::arg("v"),
-               py::arg("to_change"), py::arg("new_conf"), R"EOF(
+          .def(
+              "update_conf",
+              [](const AbstractHilbert &self, Eigen::Ref<Eigen::VectorXd> v,
+                 PyCArray<int> tochange, PyCArray<double> newconf) {
+                if (tochange.ndim() > 1 || newconf.ndim() > 1) {
```

Comment:
making these checks at every function call can be rather expensive, are we sure we don't want to put those only in debug mode?

Reply:
Since we know these are 1d arrays, we should maybe use this https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html#direct-access for performance reasons

Reply:
I would expect the overhead of these checks to be insignificant compared to the function call overhead when are calling C++ from Python, which we have here anyway.

The access to the data within `AbstractHilbert::UpdateConf` is done via the `span`s constructed below, which omit bounds checking (just as the direct access functions you mentioned would).